lib/Sema/CSApply.cpp

```diff
@@ -7527,6 +7527,39 @@ Expr *ConstraintSystem::applySolution(Solution &solution, Expr *expr,
     return nullptr;
   }
 
+  for (auto &e : solution.Conformances) {
+    const auto &conformance = e.second;
+
+    auto anchor = expr;
+    if (auto *locator = e.first)
+      anchor = locator->getAnchor();
+
+    if (conformance.isAbstract())
+      continue;
+
+    auto *concrete = conformance.getConcrete();
+    // If conformance check is not yet complete let's re-check using 'Used'
+    // to require conformance check to happen.
+    if (concrete->getState() != ProtocolConformanceState::Complete) {
+      if (auto conformance = TC.conformsToProtocol(
+              concrete->getType(), concrete->getProtocol(), DC,
+              ConformanceCheckFlags::InExpression |
+                  ConformanceCheckFlags::Used)) {
+        if (conformance->isAbstract())
+          continue;
+
+        concrete = conformance->getConcrete();
+      }
+    }
+
+    if (concrete->isComplete() && concrete->isInvalid()) {
```

Comment:
The intent here is to make sure the conformance is marked `Used`, so it's completed in case something later in the pipeline (SILGen, SIL optimizers, IRGen) needs it... because once the type checker is gone, these conformances can no longer be completed. We have a bunch of ad hoc logic in CSApply that marks conformances used... this is more likely to catch all of them.